src/nvim/eval_defs.h

```diff
@@ -16,39 +16,52 @@ typedef double float_T;
 typedef struct listvar_S list_T;
 typedef struct dictvar_S dict_T;
 
-/*
- * Structure to hold an internal variable without a name.
- */
+/// Special variable values
+typedef enum {
+  kSpecialVarFalse,  ///< v:false
+  kSpecialVarTrue,   ///< v:true
+  kSpecialVarNull,   ///< v:null
+} SpecialVarValue;
+
+/// Variable lock status for typval_T.v_lock
+typedef enum {
+  VAR_UNLOCKED = 0,  ///< Not locked.
+  VAR_LOCKED,        ///< User lock, can be unlocked.
+  VAR_FIXED,         ///< Locked forever.
+} VarLockStatus;
+
+/// VimL variable types, for use in typval_T.v_type
+typedef enum {
+  VAR_UNKNOWN = 0,  ///< Unknown (unspecified) value.
+  VAR_NUMBER,       ///< Number, .v_number is used.
+  VAR_STRING,       ///< String, .v_string is used.
+  VAR_FUNC,         ///< Function referene, .v_string is used for function name.
+  VAR_LIST,         ///< List, .v_list is used.
+  VAR_DICT,         ///< Dictionary, .v_dict is used.
+  VAR_FLOAT,        ///< Floating-point value, .v_float is used.
+  VAR_SPECIAL,      ///< Special value (true, false, null, none), .v_special
```

Comment:
20.03.2016, 16:22, "oni-link" <notifications@github.com>:In src/nvim/eval_defs.h:> +typedef enum {
> +  VAR_UNLOCKED = 0,  ///< Not locked.
> +  VAR_LOCKED,        ///< User lock, can be unlocked.
> +  VAR_FIXED,         ///< Locked forever.
> +} VarLockStatus;
> +
> +/// VimL variable types, for use in typval_T.v_type
> +typedef enum {
> +  VAR_UNKNOWN = 0,  ///< Unknown (unspecified) value.
> +  VAR_NUMBER,       ///< Number, .v_number is used.
> +  VAR_STRING,       ///< String, .v_string is used.
> +  VAR_FUNC,         ///< Function referene, .v_string is used for function name.
> +  VAR_LIST,         ///< List, .v_list is used.
> +  VAR_DICT,         ///< Dictionary, .v_dict is used.
> +  VAR_FLOAT,        ///< Floating-point value, .v_float is used.
> +  VAR_SPECIAL,      ///< Special value (true, false, null, none), .v_special
none does not seem to be a value of VAR_SPECIAL. Fixed as a part of `eval: Remove v:none`. —You are receiving this because you were mentioned.Reply to this email directly or view it on GitHub 